extensions/security/deployment/src/main/java/io/quarkus/security/deployment/SecurityConfig.java

```diff
@@ -13,6 +13,12 @@
 @ConfigRoot(phase = ConfigPhase.BUILD_TIME)
 public final class SecurityConfig {
 
+    /**
+     * Whether authorization is enabled in dev mode or not. In other launch modes authorization is always enabled.
+     */
+    @ConfigItem(name = "authorization.enabled", defaultValue = "true")
+    public boolean authorizationEnabled;
```

Comment:
Not sure whether the config key should include a segment which makes it clear that it applies to dev mode only. Also wasn't sure whether to create a new `authorization` "namespace" or just call the flag `authorization-enabled`.

Reply:
@knutwannheden I think it makes sense to qualify it is dev mode related, may be `devModeAuthorizationEnabled`, @stuartwdouglas, Stuart, what do you think ?

Reply:
This should definitely have something in the name that specifies that it is dev mode only, otherwise confusion will result.

Reply:
Renaming to `quarkus.security.authorization.enabled-in-dev-mode`. I found a few other config keys with the `-in-dev-mode` suffix.